content/blog/how-tikv-accesses-data.md

```diff
@@ -0,0 +1,263 @@
+    number="" >}}
+
+TiKV uses the Raft consensus algorithm to ensure data safety and consistency. By default, three replicas are supported, and these three replicas form a Raft Group.
+
+When the client needs to write some data, it sends the request to the Raft Leader. This process is called **Propose** in TiKV. The Leader encodes the operation into an entry and writes it into its own **Raft Log**. This is called **Append**.
+
+The Leader will also copy the entry to other Followers through the Raft algorithm. This is called **Replicate**. After the Follower receives the entry, it will also perform an **Append** operation, and in the meantime inform the Leader that the Append is successful.
+
+When the Leader finds that the entry has been appended by majority of nodes, it considers that the entry Committed. Then, it can decode the operations in the entry, execute them, and apply them to the state machine. This is called **Apply**.
+
+TiKV supports a feature called Lease Read. For Read requests, they can be sent directly to the Leader. If the Leader determines that its lease has not expired, it directly provides the Read service without having to go through the Raft process. If the lease has expired, the Leader will be forced to go through the Raft process to renew the lease before it provides the Read service.
+
+
+### Multi Raft
+
+{{< figure
+    src="/img/blog/how-tikv-accesses-data/multiraft.png"
+    caption="Multi Raft"
+    number="" >}}
+
+Because one Raft Group only processes a limited amount of data, we split the data into multiple Raft Groups, each of which correspond to a Region. The way of splitting is to slice by range. That is, we sort the keys of the data in byte order to form an infinite sorted map, and then slice it into a segments of continuous key range. Each key range is treated as a Region. The range of the Region uses the front-end and back-open mode [start, end). The end key of the previous region is the start key of the next region.
+
+Regions in TiKV have a maximum size limit. When this threshold is exceeded, it splits into two Regions, such as [a, b)-> [a, ab) + [ab, b). Conversely, if there is very little data in the Region, it will be merged with the neighboring Regions to form a larger Region, such as `[a, ab) + [ab, b)-> [a, b)`.
+
+### Percolator
+
+For the same Region, we can guarantee the consistency of the key operations  through the Raft consensus protocol. But if we want to operate data that falls on different regions at the same time, we need distributed transactions to ensure the consistency of operations.
+
+The most common method of distributed transactions is two-phase commit, also known as 2PC. TiKV draws its inspirations of 2PC implementation from Google's Percolator, with some practical optimizations to provide the distributed transaction support. The underlying mechanism of Percolator is rather complicated. For illustration purposes, we highlight the following few points:
```

Comment:
```suggestion
The most common method of distributed transactions is two-phase commit, also known as 2PC. TiKV draws its inspirations of 2PC implementation from Google's Percolator, with some practical optimizations to provide the distributed transaction support. For illustration purposes, let's highlight a few points of the mechanism:
```

Avoid hand-wavy "It's complicated" sentences, just focus on what you will tell them.

Reply:
Good point!